Sources_v3/StreamChatUI/ChatChannel/MessageComposerInputAccessoryViewController.swift

```diff
@@ -7,20 +7,57 @@ import UIKit
 
 open class MessageComposerInputAccessoryViewController<ExtraData: UIExtraDataTypes>: UIInputViewController,
     UIConfigProvider,
+    Customizable,
+    AppearanceSetting,
     UITextViewDelegate,
     UIImagePickerControllerDelegate,
     UIDocumentPickerDelegate,
     UINavigationControllerDelegate {
+    // MARK: - Underlying types
+    
+    public enum State {
+        case initial
+        case empty(isEmpty: Bool)
+        case slashCommand(MessageInputSlashCommandView.Command)
+        case suggestions
+        case forceShrinkedInput
+        case reply
+        case edit
+    }
+    
     // MARK: - Properties
     
     var controller: _ChatChannelController<ExtraData>!
     
+    public var state: State = .initial {
+        didSet {
+            updateContent()
+        }
+    }
+    
+    public var replyMessage: _ChatMessage<ExtraData>? {
+        didSet {
+            state = .reply
+        }
+    }
+    
+    public var messageToEdit: _ChatMessage<ExtraData>? {
+        didSet {
+            state = .edit
+        }
+    }
```

Comment:
This looks extremely hacky. Why not extend the `State` enum with associated values for the messages?
```swift
case reply(message:  _ChatMessage<ExtraData>)
case edit(message:  _ChatMessage<ExtraData>)
```

Reply:
This `states` will definitely have some changes soon so I decided to leave that to actual implementation of replies and editing.